test/e2e/annotations_test.go

```diff
@@ -431,9 +433,80 @@ metadata:
 	})
 }
 
+func TestAdoptionOfResourcesWithVersionedAnn(t *testing.T) {
+	env := BuildEnv(t)
+	logger := Logger{}
+	kubectl := Kubectl{t, env.Namespace, logger}
+	kapp := Kapp{t, env.Namespace, env.KappBinaryPath, logger}
+
+	yaml := `
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: config
+  annotations:
+    kapp.k14s.io/versioned: ""
+data:
+  key1: val1
+`
+
+	name := "test-adoption-of-res-with-ver-ann"
+	cleanUp := func() {
+		kapp.Run([]string{"delete", "-a", name})
+	}
+
+	cleanUp()
+	defer cleanUp()
+
+	logger.Section("Kapp should adopt already deployed versioned resource through kubectl", func() {
+		out, _ := kubectl.RunWithOpts([]string{"apply", "-f", "-", "-o", "json"},
+			RunOpts{IntoNs: true, StdinReader: strings.NewReader(yaml)})
+
+		respKubectl := appv1alpha1.App{}
```

Comment:
not sure why we are deserializing into App CR. we should use corev1.ConfigMap type (as that what that is).

Reply:
Yeah my bad. Done with the changes.